src/devices/bus/pce/pce_scdsys.h

```diff
@@ -0,0 +1,84 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli, Wilbert Pol
+#ifndef MAME_BUS_PCE_PCE_SCDSYS_H
+#define MAME_BUS_PCE_PCE_SCDSYS_H
+
+#pragma once
+
+#include "pce_slot.h"
+
+
+// ======================> pce_cdsys3_base_device
+
+class pce_cdsys3_base_device : public device_t
+{
+public:
+	// construction/destruction
+	pce_cdsys3_base_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// configuration
+	void set_region(bool region) { m_region = region; }
+
+	// reading and writing
+	uint8_t ram_r(offs_t offset);
+	void ram_w(offs_t offset, uint8_t data);
+	uint8_t register_r(offs_t offset);
+
+protected:
+	// device-level overrides
+	virtual void device_start() override;
+
+private:
+	std::unique_ptr<uint8_t[]> m_ram; // internal RAM
+	bool m_region = false; // Cartridge region
+};
```

Comment:
Is there really a benefit of making this a full device rather than just a helper class that can be instantiated as a simple member at most?  The only thing it really implements is the “register” read stuff which is fairly trivial.

The RAM can be done using a `memory_share_creator` and installed with `install_ram`.